BrainPortal/lib/boutiques_input_copier.rb

```diff
@@ -0,0 +1,243 @@
+#       "optional": true,
+#       "type": "File",
+#       "command-line-flag": "-i",
+#       "value-key": "<-i input>"
+#     },
+#
+# and in the `cbrain:integrator_modules` section look like:
+#
+#     "BoutiquesInputCopier": {
+#       "parent_input_id": {"default": boolean, "hide": boolean, "description": string},
+#      },
+#
+# The final command line will be:
+#
+#     apptool <-i input_copy>...
+#
+#
+module BoutiquesInputCopier
+
+  # Note: to access the revision info of the module,
+  # you need to access the constant directly, the
+  # object method revision_info() won't work.
+  Revision_info=CbrainFileRevision[__FILE__] #:nodoc:
+
+
+  ############################################
+  # Portal Side Modifications
+  ############################################
+
+  # Adjust the description for the input with the mention of
+  # the fake parent directory information
+  #
+  # For each input in BoutiquesInputCopier section create a new entry in the descriptor
+  # that will be a flag to choose if the input will be copied if the checkbox is selected
+  def extension_of_descriptor(descriptor) #:nodoc:
+    parent_input_ids = descriptor.custom_module_info('BoutiquesInputCopier')
+
+    inputs      = descriptor["inputs"]
+
+    # In parent_input_ids, select all non hidden inputs
+    parent_input_ids = parent_input_ids.select { |_, config| !config["hide"] }
+    copied_info = fill_copied_info(parent_input_ids, descriptor, inputs)
+
+    # Add the new inputs to the descriptor and to the group if needed
+    extracted_idx = copied_info.values.map { |info| info[:input_idx] }.sort.reverse
+
+    extracted_idx.each do |index|
+      # In copied_info, find the one with input_idx == index
+      copied_info.each do |parent_inputid, info|
+        next if info[:hide]
+        next if info[:input_idx] != index
+
+        # Insert the new input in the descriptor
+        new_input = info[:new_input]
+        inputs.insert(index, new_input)
+
+        # Insert the new input in the group
+        next if !info[:member_info]
+        group        = descriptor.group_by_id(info[:member_info][:id])
+        members      = group["members"]
+        new_input_id = new_input["id"]
+        next if members.include?(new_input_id)
+        members.insert(info[:member_info][:idx], new_input_id)
+      end
+    end
+
+    descriptor
+  end
+
+  def descriptor_for_before_form #:nodoc:
+    descriptor = super.dup
+    extension_of_descriptor(descriptor)
+  end
+
+  def descriptor_for_form #:nodoc:
+    descriptor = super.dup
+    extension_of_descriptor(descriptor)
+  end
+
+  def descriptor_for_after_form #:nodoc:
+    descriptor = super.dup
+    extension_of_descriptor(descriptor)
+  end
+
+  ############################################
+  # Bourreau (Cluster) Side Modifications
+  ############################################
+
+  # For input in +BoutiquesInputCopier+ section,
+  def setup #:nodoc:
+    descriptor = self.descriptor_for_setup
```

Comment:
Why set this variable here before invoking super two lines below? Run the super() first.